projecttacoma/fqm-testify
# Render clause coverage HTML

# Summary
Adds a button and new page to show the clause coverage HTML after running population calculation.

## New behavior
After the user clicks on the “Calculate Population Results” button and the calculation has finished, a new button will appear with the text “See Clause Coverage.” When clicked, the user will be brought to a new page that shows the clause coverage HTML and the clause coverage percentage. When finished viewing the HTML, the user can click the back button to return to the main page.

## Code changes
* `BackButton` component that takes the user back to the previous page
* `PopulationCalculation` component changes
    * Set `calculateClauseCoverage` to `true` within the calculation options so that we can retrieve the clause coverage HTML from the fqm-execution results
    * New button that appears on successful calculation that links to the new `[measureid]/coverage.tsx` page and passes the clause coverage HTML to the new page
* New helper in `testHelpers.tsx` for creating the mock router used to test routing to the new page
* Unit tests for coverage page, and updated unit tests for population calculation coverage button

# Testing Guidance
**NOTE: the fqm-execution path was changed in `package.json` to get access to the coverage HTML code**. We need to include the clause coverage updates in the latest npm release of fqm-execution.
When testing, make sure to run `npm install` to get the clause coverage changes from fqm-execution. Then, try running `npm run check` to check that the unit tests, lint, and prettier all pass. Try running `npm ci` and check that it executes successfully.

Run the unit tests that test the functionality of the new route and clause coverage HTML rendering.

Try navigating to the clause coverage page in the app by doing the following:
* Upload a measure, select the appropriate measurement period, and import/create test cases
* Click on “Calculate Population Results”
* Check for a new button to appear that says “Show Clause Coverage,” then click on it
* You should be brought to the new page. Check that the clause coverage HTML gets rendered, and that all the necessary information is available for the user (the measure id, clause coverage percentage, etc. Let me know if you think anything else should be on the page, like the patient names used in the calculation or any other information from the calculation)
* Run through the process again, but try adding/removing patients for certain populations. You should see the highlighted clauses change, and the percentage should change as well (for example, running calculation with just denominator patients versus running calculation with both numerator and denominator patients)
